contracts/BIFI/infra/BeefyFeeConfigurator.sol

```diff
@@ -0,0 +1,143 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+interface IStrategy {
+    function vault() external view returns (address);
+}
+
+contract BeefyFeeConfigurator is OwnableUpgradeable {
+
+    struct FeeCategory {
+        uint256 total;
+        uint256 beefy;
+        uint256 call;
+        uint256 strategist;
+        string label;
+        bool active;
+    }
+
+    address public keeper;
+    uint256 public totalLimit;
+    uint256 constant DIVISOR = 1 ether;
+
+    mapping(address => uint256) public vaultFeeId;
+    mapping(uint256 => FeeCategory) internal feeCategory;
+
+    event UpdateVault(address indexed vault, uint256 indexed id);
+    event SetFeeCategory(
+        uint256 indexed id,
+        uint256 total,
+        uint256 beefy,
+        uint256 call,
+        uint256 strategist,
+        string label,
+        bool active
+    );
+    event SetTotalLimit(uint256 totalLimit);
+    event Pause(uint256 indexed id);
+    event Unpause(uint256 indexed id);
+
+    function initialize(
+        address _keeper,
+        uint256 _totalLimit
+    ) public initializer {
+        __Ownable_init();
+
+        keeper = _keeper;
+        totalLimit = _totalLimit;
+    }
+
+    // checks that caller is either owner or keeper
+    modifier onlyManager() {
+        require(msg.sender == owner() || msg.sender == keeper, "!manager");
+        _;
+    }
+
+    // fetch fees for the connected vault when called by a strategy
+    function getFees() external view returns (FeeCategory memory) {
+        address vault = IStrategy(msg.sender).vault();
```

Comment:
it should be explicit `getFees(address)`. no need to make additional external call, and maybe more important to not hardcode knowledge here that there is a strat, that strat has a vault etc

Reply:
Implemented this.